app/models/symphony_curr_loc_request.rb

```diff
@@ -0,0 +1,53 @@
+require 'json'
+
+# API for retrieving the current location from a barcode via Symphony Web Services
+class SymphonyCurrLocRequest
+  include ActiveModel::Model  # allows initialization with a hash of attributes
+
+  attr_accessor :barcode
+
+  def current_location
+    return '' if json.blank?
+    json['fields']['currentLocation']['key'] || ''
+  end
+
+  private
+
+  def response
+    @response ||= Faraday.get do |req|
+      req.url url
+      # need the below for symws catalog/item/barcode
+      req.headers['x-sirs-clientID'] = 'DS_CLIENT'
+      req.headers['sd-originating-app-id'] = 'requests'
+      req.headers['SD-Preferred-Role'] = 'GUEST'
+    end
+    return empty_response(@response.body) unless @response.success?
+    @response
+  rescue Faraday::Error::ConnectionFailed => e
+    empty_response(e)
+  end
+
+  def json
+    return {} unless response.success?
+    @json ||= JSON.parse(response.body)
+  rescue JSON::ParserError
+    {}
```

Comment:
similarly, not sure if this should return empty hash if there was an error ...   esp. as `currrent_location` method above kinda barfs.   Maybe it should say .empty?  instead of .blank?